app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js

```diff
@@ -1223,7 +1223,14 @@ define([
                 }
 
                 imagesToUpdate = this._setImageIndex(imagesToUpdate);
-                gallery.updateData(imagesToUpdate);
+                if (gallery === undefined) {
```

Comment:
Such check is not correct, please use `typeof` instead.

Reply:
I will update the code.
Please review It.